### ISSUE TITLE
Use consistent-env to detect PATH

### ISSUE THREAD

**Issue by steelbrain** [aki77/atom-env-from-shell#2]
[`consistent-env`](https://www.npmjs.com/package/consistent-env) is pretty battle tested and will make your package lightweight

**Comment by aki77** [comment]
Thanks for your advice.
Does it work with `csh` and `fish`?

**Comment by steelbrain** [comment]
It doesn't fix the env but it does return what's available. If you would be willing to add your expertise to it, it'll have an impact on a broader audience

**Comment by steelbrain** [comment]
I've added support for `fish` in the latest version of the package

**Comment by aki77** [comment]
Is this no good?

``` javascript
export function getCommand() {
  let command = process.env.SHELL
  let parameters = ['-i', '-c', 'env;exit']
  let options = {timeout: 3000, encoding: 'utf8'}

  return {command, parameters, options}
}
```

**Comment by steelbrain** [comment]
`-i` makes bash and zsh and all the other major shells hang forever, It's a known bug that hopefully isn't going to be fixed for the next few centuries.

That code you mentioned above will work for some use cases, but for packages that are executed with `nohup` it'll hang them in a deadlock and they'll mysteriously use a lot of CPU. Atom is run with `nohup` on linux so that's where it breaks.

I had made a repo with a repro when I encountered it https://github.com/steelbrain/nohup-shell-hanging-bug
